src/csr.hpp

```diff
@@ -202,95 +180,59 @@ class MatrixOperator
 
     if (transpose)
     {
-      int num_rows = _row_map->size_local();
+      int num_rows = _A->index_map(0)->size_local();
       dim3 block_size(256);
       dim3 grid_size((num_rows + block_size.x - 1) / block_size.x);
       x.scatter_fwd_begin(get_pack_fn<T>(512),
                           [](auto&& x) { return x.data().get(); });
       impl::spmvT_impl<T><<<grid_size, block_size, 0, 0>>>(
-          num_rows, _values.data().get(), _row_ptr.data().get(),
-          _off_diag_offset.data().get(), _cols.data().get(), _x, _y);
+          num_rows, _A->values().data().get(), _A->row_ptr().data().get(),
+          _A->off_diag_offset().data().get(), _A->cols().data().get(), _x, _y);
       check_device_last_error();
       x.scatter_fwd_end(get_unpack_fn<T>(512, 1));
 
       impl::spmvT_impl<T><<<grid_size, block_size, 0, 0>>>(
-          num_rows, thrust::raw_pointer_cast(_values.data()),
-          thrust::raw_pointer_cast(_off_diag_offset.data()),
-          thrust::raw_pointer_cast(_row_ptr.data()) + 1,
-          thrust::raw_pointer_cast(_cols.data()), _x, _y);
+          num_rows, thrust::raw_pointer_cast(_A->values().data()),
+          thrust::raw_pointer_cast(_A->off_diag_offset().data()),
```

Comment:
`thrust::raw_pointer_cast(_A->off_diag_offset().data())` -> `_A->off_diag_offset().data().get()`

Reply:
I was a bit confused about the mix of `.get()` and `thrust::raw_pointer_cast` in the code... I guess `.get()` is cleaner...